File: analytics/metrics-platform/src/test/java/org/wikimedia/metrics_platform/event/EventFixtures.java
package org.wikimedia.metrics_platform.event;

import static org.wikimedia.metrics_platform.context.DataFixtures.getTestClientData;
import static org.wikimedia.metrics_platform.event.EventProcessed.fromEvent;

import java.util.Arrays;
import java.util.List;

import org.wikimedia.metrics_platform.context.ClientData;
import org.wikimedia.metrics_platform.context.PageData;
import org.wikimedia.metrics_platform.context.PerformerData;

public final class EventFixtures {

    private EventFixtures() {
        // Utility class, should never be instantiated
    }
    public static Event minimalEvent() {
        return new Event("test_schema", "test_stream", "test_event");
    }

    public static EventProcessed minimalEventProcessed() {
        EventProcessed eventProcessed = fromEvent(minimalEvent());
        eventProcessed.setClientData(getTestClientData());
        return eventProcessed;
    }

    public static EventProcessed getEvent() {
        Event event = new Event("test/event", "test.event", "testEvent");
        ClientData clientData = new ClientData();
        clientData.setPageData(PageData.builder().id(1).namespaceName("Talk").build());

        event.setClientData(clientData);
        EventProcessed eventProcessed = fromEvent(event);
        eventProcessed.setPerformerData(
                PerformerData.builder()
                        .groups(Arrays.asList("user", "autoconfirmed", "steward"))
                        .isLoggedIn(true)
                        .build()
        );
        return eventProcessed;
    }

    public static EventProcessed getEvent(
            Integer id,
            String namespaceName,
            List<String> groups,
            boolean isLoggedIn,
            String editCount
    ) {
        Event event = new Event("test/event", "test.event", "testEvent");
        ClientData clientData = new ClientData();
        clientData.setPageData(PageData.builder().id(id).namespaceName(namespaceName).build());

        event.setClientData(clientData);
        EventProcessed eventProcessed = fromEvent(event);
        eventProcessed.setPerformerData(
                PerformerData.builder()
                        .groups(groups)
                        .isLoggedIn(isLoggedIn)
                        .build()
        );
        return eventProcessed;
    }
}

File: analytics/testkitchen/src/test/java/org/wikimedia/testkitchen/event/EventFixtures.kt
package org.wikimedia.testkitchen.event

import org.wikimedia.testkitchen.context.ClientData
import org.wikimedia.testkitchen.context.InteractionData
import org.wikimedia.testkitchen.context.PageData
import org.wikimedia.testkitchen.context.PerformerData
import java.time.ZoneId
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter

object EventFixtures {
    private val dt get() = DateTimeFormatter.ISO_DATE_TIME.format(ZonedDateTime.now(ZoneId.of("Z")))

    fun minimalEvent(clientData: ClientData = ClientData()): Event {
        return Event("test_schema", "test_stream", dt, clientData, InteractionData())
    }

    fun getEvent(
        id: Int?,
        namespaceName: String?,
        groups: List<String>,
        isLoggedIn: Boolean,
        editCount: String?
    ): Event {
        return Event("test/event", "test.event", dt,
            ClientData(
                pageData = PageData().also {
                    it.id = id
                    it.namespaceName = namespaceName
                },
                performerData = PerformerData().also {
                    it.groups = groups
                    it.isLoggedIn = isLoggedIn
                }
            ),
            InteractionData()
        )
    }
}
